src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -683,12 +683,14 @@ private string GeneratePropMetadataInitFunc(TypeGenerationSpec typeGenerationSpe
                         ? @$"jsonPropertyName: ""{memberMetadata.JsonPropertyName}"""
                         : "jsonPropertyName: null";
 
-                    string getterNamedArg = memberMetadata.CanUseGetter
+                    string getterNamedArg = memberMetadata.CanUseGetter &&
+                        memberMetadata.DefaultIgnoreCondition != JsonIgnoreCondition.Always
```

Comment:
Curious, how does the presence of ObsoleteAttribute influence the value of `memberMetadata.DefaultIgnoreCondition`?

Reply:
In the issue this generated code is shown:
```cs
    getter: static (obj) => ((global::MyLibrary.Models.ClassA)obj).Foo,
    setter: static (obj, value) => ((global::MyLibrary.Models.ClassA)obj).Foo = value!,
```

Since the property `Foo` is used but obsolete it generates a compiler warning.